resources/shaders/rt_reflection_shadows.hlsl

```diff
@@ -192,13 +192,106 @@ void RaygenEntry()
 	// Compute the initial ray cone from the gbuffers.
  	RayCone cone = ComputeRayConeFromGBuffer(sfhit, 1.39626, DispatchRaysDimensions().y);
 	
-	float shadow_result = DoShadowAllLights(wpos + normal * EPSILON, 0, rand_seed);
+	// Get shadow factor
+	float4 shadow_result = DoShadowAllLights(wpos + normal * EPSILON, V, normal, metallic, roughness, albedo, 0, rand_seed);
 
 	// Get reflection result
 	float3 reflection_result = clamp(DoReflection(wpos, V, normal, rand_seed, depth, cone), 0, 100000);
 
 	// xyz: reflection, a: shadow factor
-	output_refl_shadow[DispatchRaysIndex().xy] = float4(reflection_result.xyz, shadow_result);
+	output_refl_shadow[DispatchRaysIndex().xy] = float4(reflection_result.xyz, shadow_result.w);
+}
+
+[shader("raygeneration")]
+void ShadowRaygenEntry()
```

Comment:
Since its now 2 passes you can split this up into 2 files. and move duplicate code into functions.

Reply:
Emilio's shader refactoring takes care of this.